akka-http-core/src/main/scala/akka/http/impl/engine/parsing/BodyPartParser.scala

```diff
@@ -124,7 +135,9 @@ private[http] final class BodyPartParser(
         try {
           @tailrec def rec(index: Int): StateResult = {
             val needleEnd = boyerMoore.nextIndex(input, index) + needle.length
+            val needleLFEnd = boyerMoore.nextIndex(input, index) + needleLF.length
```

Comment:
`boyerMoore` is dependent on `needle` so I wonder if we'd also need a `boyerMooreLF` to correctly calculate the next index here.